### ISSUE TITLE
datasets split_train_val() parameter error

### ISSUE THREAD

**Issue by taoshende** [Polygl0t/llm-foundry#57]
### System Info

Bender, Python environment created by `distributed/slurm/create_venv_bender.sh` 

### Information

- [x] An existing script in the repository (e.g. under `distributed/`, `data/`, `evals/`)
- [ ] My own modified or new scripts

### Area

- [ ] Pretraining (`distributed/`)
- [ ] Post-training / alignment (`alignment/`)
- [x] Data processing / tokenization (`data/`): `data/filters/train_annotator.py`
- [ ] Evaluation (`evals/`)
- [ ] Synthetic data (`synthetic/`)
- [ ] Model merging (`merge/`)
- [ ] Tokenizer (`tokenizer/`)
- [ ] Other / tools (`tools/`)

### Reproduction

1. Create dataset `data.jsonl` of size `n`, such that `n mod 10 != 0`.
2. Run `data/filters/train_annotator.py` with parameter `--train_dataset_dir data.jsonl`.

#### Encountered Error

We encounter a parameter type error while creating the split into train and test set. We set the size of the test set as `n * 0.1` which leads to a non-integer number. However, the parameter `test_size` in the function `datasets.Dataset.train_test_split()` only accepts either type `float` between `0.0` and `1.0` or type `int` . See [docs](https://huggingface.co/docs/datasets/main/en/package_reference/main_classes#datasets.Dataset.train_test_split).
```bash
  # Split the dataset into train and test sets.
  dataset = dataset.train_test_split(
      test_size=min(
          args.test_size, len(dataset) * 0.1       # <- this can be non-integer > 1.0
      ),  # Ensure test_size doesn't exceed dataset size
      seed=args.seed,
      stratify_by_column=args.target_column,
  )
```

#### Proposed Fix
Cast `len(dataset) * 0.1` to `int`:
```bash
  dataset = dataset.train_test_split(
      test_size=min(
          args.test_size, int(len(dataset) * 0.1)
      ),
     ...,
  )
```

### Expected behavior

We expect the training to run uninterrupted.

**Comment by Nkluge-correa** [comment]
Good catch! This is a bug. I'll review it right away.
